### PR TITLE
Editable timestamps and mobile horizontal scroll

### LINKED ISSUE #8: Display incomplete on mobile
First of all: This is such a cool plugin. I use it A LOT to track the time spent on writing projects and divide time into tasks like research, drafting, writing and so forth.

There is one glitch:
I am using the Android client on my smartphone and tablet. On the smaller display, I cannot get to the edit handles on the right side as they are cut off. For some reason, I cannot even scroll there. the only solution is to enable rotation and go into widescreen mode.

EDIT:
I just checked: The same happens on desktop once the note width is narrow enough. At first, the fields reflow nicely (GREAT!!!) but once they are still too wide, the block gets cut off:

Maybe the fields could vanish one by one. If I may, I would suggest this order of dissappearing:
1. End time
2. Start time
3. Duration

This way, the timer could always be read and edited.

### LINKED ISSUE #9: Make Timecode editable in readable format
Sometimes I forget to switch the timer off. I then usually have a rough idea of how much time I spent on this particular chunk of work. So I then edit the codeblock to something close to this.

The problem I have: The code looks like this:
"startTime":1666021639,"endTime":1666027580}

This makes it very hard to adjust the time, It would be fantastic if the time parameters could be made editable like the segment name. Switching fields could easily be done by using the tab key.

### PR DESCRIPTION
**Pull request by bagaag**
This PR resolves two open issues. I've tested these on both the Linux and Android clients.

**Ability to edit timestamps of entries**

Fixes #9

I extended the logic used for the editable entry label to add the ability to edit time stamps as well. Doing so resulted in a lot of repeated code, so I consolidated management of editable fields into a class. 

Editable time stamps are formatted using the user's setting, which requires that format be specific enough to convert back into a timestamp. This could cause confusion if a user sets a format that does not convert back into a timestamp. The alternative would be to use a fixed format or a separate setting for editing timestamps.

**Entry buttons cut off in mobile**

Fixes #8

I encountered this bug while testing the change on mobile. This is a simple fix by making the container horizontally scrollable if needed.

### CODE REVIEW

File: src/tracker.ts [hunk 1]
```diff
@@ -268,18 +273,48 @@ function createTableSection(entry: Entry, settings: SimpleTimeTrackerSettings):
...
+class EditableField {
+    cell: HTMLTableCellElement;
+    label: HTMLSpanElement;
+    box: TextComponent;
+    constructor(row: HTMLTableRowElement, indent: number, value: string) {
+        this.cell = row.createEl("td");
+        this.label = this.cell.createEl("span", { text: value });
+        this.label.style.marginLeft = `${indent}em`;
+        this.box = new TextComponent(this.cell).setValue(value);
+        this.box.inputEl.width = 10;
```
Comment (Ellpeck) [inline src/tracker.ts:285]: Why are you setting the width to 10 here?

File: styles.css [hunk 2]
```diff
@@ -62,3 +66,8 @@
 .simple-time-tracker-table .clickable-icon {
     display: inline;
 }
+
+.simple-time-tracker-input {
+    max-width: 150px;
+    min-width: 100px;
```
Comment (Ellpeck) [inline styles.css:72]: What are these new style settings for?

**Review by Ellpeck — COMMENTED** [review]
Hi, thanks so much for this pull request! I've asked some questions before I'm ready to merge the request, but it looks pretty good so far all in all. 

I'm a little worried about the editable timestamp format issue you described, especially since I, for example, don't include the year in my time tracker time format. Maybe it would make most sense to lock the editable time format to the default display setting of `YY-MM-DD hh:mm:ss`, rather than also basing it on the display setting.

**Comment by bagaag** [comment]
The width = 10 was not meant to be left there. I've removed it.

The input style was needed to achieve consistent formatting on desktop and
mobile (Android). Without that style, the boxes were way too wide on both
platforms and became too narrow to edit on Android.

As for the format setting, I actually just use "HH:mm:ss" because my
entries never span more than 1 day, so I only care about relative time. It
works for that purpose. The default setting of "YY-MM-DD hh:mm:ss" is
actually problematic because "hh" is 12 hrs, not 24. We'd need to add an
"a" (AM/PM) to that format for it to work properly. Let me know if you
prefer a hard-coded format of "YY-MM-DD HH:mm:ss" or "YY-MM-DD hh:mm:ss a"
and I can make that change. Personally, I like "HH:mm:ss" as it's smaller
and works if entries don't span dates. But letting users set this could
definitely cause confusion.

**Comment by Ellpeck** [comment]
Thanks for the explanations!!

I think hardcoding a 24 hour format with four year digits would make most sense, so `YYYY-MM-DD HH:mm:ss`. I hadn't actually noticed the HH/hh discrepancy in the original code before, so that may also be something that needs fixing at some point!

**Comment by bagaag** [comment]
Sounds good. I should be able to make that change in the next couple of days. 

If you like, I can make the change of hh to HH in settings.ts while I'm at it.

**Comment by Ellpeck** [comment]
> If you like, I can make the change of hh to HH in settings.ts while I'm at it.

Yes, that would be lovely! Thanks.

**Comment by Ellpeck** [comment]
Looks great, thanks so much again! 

I'll merge this as soon as I get a chance to work on the plugin again, so that I can get it properly right away.

**Review by Ellpeck — APPROVED** [review]

**Pull request merged**